packages/pigeon/example/app/ios/Runner/AppDelegate.swift

```diff
@@ -25,6 +25,10 @@ private class PigeonApiImplementation: ExampleHostApi {
     }
     completion(.success(true))
   }
+
+  func sendMessageModernAsync(message: MessageData) async -> Bool {
```

Comment:
I just looked at the code again - it looks like the contract of pigeon is that the API (e.g. `sendMessage`) should be called on main thread. If that's the contract we want to preserve, we should mark the async API as main actor. 

```
@MainActor
func sendMessageModernAsync(message: MessageData) async -> Bool {}
```

Reply:
The only thing that should be called on the main thread is dispatching messages via channel

https://docs.flutter.dev/platform-integration/platform-channels#jumping-to-the-main-thread-in-ios

Reply:
iiuc both way of communication needs to be on main thread. 

The link you add here is calling from platform to dart side. This is the opposite side communication, which needs to be on main as well.

Reply:
Actually it is the same from the point of dispatching reply via channel, be it call from platform to dart or replying to call from dart